sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ier12-integration/_tests_/e2e/ier12-factory.ts

```diff
@@ -0,0 +1,342 @@
+  db: E2EDataSources,
+  testInputAssessment: IER12Assessment,
+  assessment: StudentAssessment,
+  studyStartDate: string,
+  studyEndDate: string,
+  referenceSubmission: Date,
+): Promise<StudentAssessment> {
+  assessment.triggerType = testInputAssessment.triggerType;
+  // Simulates that the assessment date was produced one day after the reference date
+  // in case the assessmentDate was not provided. It helps creating different dates
+  // which allows a better data assertion.
+  assessment.assessmentDate =
+    testInputAssessment.assessmentDate ?? addDays(1, referenceSubmission);
+  assessment.assessmentData =
+    testInputAssessment.assessmentData as FullTimeAssessment;
+  assessment.workflowData = testInputAssessment.workflowData;
+  // Schedules and awards mapping.
+  const [firstDisbursement, secondDisbursement] =
+    assessment.disbursementSchedules;
+  const [firstDisbursementTestInput, secondDisbursementTestInput] =
+    testInputAssessment.disbursementSchedules;
+  mapTestInputToDisbursementAndAwards(
+    firstDisbursement,
+    firstDisbursementTestInput,
+    studyStartDate,
+    referenceSubmission,
+  );
+  if (secondDisbursement) {
+    // Calculate the middle date between offering start/end dates.
+    const midOfferingDateDays =
+      dateDifference(studyEndDate, studyStartDate) / 2;
+    const midOfferingDate = addDays(midOfferingDateDays, studyStartDate);
+    mapTestInputToDisbursementAndAwards(
+      secondDisbursement,
+      secondDisbursementTestInput,
+      getISODateOnlyString(midOfferingDate),
+      referenceSubmission,
+    );
+  }
+  return db.studentAssessment.save(assessment);
+}
+
+/**
+ * Maps the test input values to the disbursements and its awards.
+ * @param disbursement disbursement being populated.
+ * @param testInputDisbursement test input to populate the disbursement and its awards.
+ * @param disbursementDate disbursement date.
+ * @param referenceSubmission optionally used for auditing dates as a fallback.
+ */
+function mapTestInputToDisbursementAndAwards(
+  disbursement: DisbursementSchedule,
+  testInputDisbursement: IER12Disbursement,
+  disbursementDate: string,
+  referenceSubmission: Date,
+): void {
+  disbursement.coeStatus = testInputDisbursement.coeStatus;
+  disbursement.disbursementScheduleStatus =
+    testInputDisbursement.disbursementScheduleStatus;
+  disbursement.disbursementDate = disbursementDate;
+
+  if (
+    disbursement.disbursementScheduleStatus === DisbursementScheduleStatus.Sent
+  ) {
+    const fallbackSentDate = new Date(disbursementDate);
+    disbursement.updatedAt =
+      testInputDisbursement.updatedAt ?? fallbackSentDate;
+    // Disbursements can be sent few days after its date.
+    disbursement.dateSent =
+      testInputDisbursement.dateSent ?? addDays(-1, fallbackSentDate);
+  } else {
+    disbursement.updatedAt =
+      testInputDisbursement.updatedAt ?? referenceSubmission;
+    disbursement.dateSent = testInputDisbursement.dateSent;
```

Comment:
How can there be a value for `dateSent`, when disbursementScheduleStatus is not Sent

Reply:
There will not be in the real world and by default it will be not provided but in case the ` testInputDisbursement.dateSent` was provided it will be used. Keep in mind that the E2E tests will produce an initial set of data that can later be modified in the test cases.